lib/diff.ml

```diff
@@ -38,18 +38,28 @@ and sig_item =
 let extract_items items =
   List.fold_left
     (fun tbl item ->
+      let visibility = Types.item_visibility item in
```

Comment:
You don't need to call this function, the visibility is part of each sig item constructor arguments so you can directly pattern match over it.